torch/testing/_internal/common_utils.py

```diff
@@ -1309,30 +1336,148 @@ def check_if_enable(test: unittest.TestCase):
         if not getattr(test, test._testMethodName).__dict__.get('slow_test', False):
             raise unittest.SkipTest("test is fast; we disabled it with PYTORCH_TEST_SKIP_FAST")
 
-# Acquires the comparison dtype, required since isclose
-# requires both inputs have the same dtype, and isclose is not supported
-# for some device x dtype combinations.
-# NOTE: Remaps bfloat16 to float32 since neither the CPU or CUDA device types
-#  support needed bfloat16 comparison methods.
-# NOTE: Remaps float16 to float32 on CPU since the CPU device type doesn't
-#   support needed float16 comparison methods.
-# TODO: Update this once bfloat16 and float16 are better supported.
-def get_comparison_dtype(a, b):
-    # TODO: update this when promote_types supports bfloat16 and/or
-    # isclose supports bfloat16.
-    a_dtype = torch.float32 if a.dtype is torch.bfloat16 else a.dtype
-    b_dtype = torch.float32 if b.dtype is torch.bfloat16 else b.dtype
-
-    compare_dtype = torch.promote_types(a_dtype, b_dtype)
-
-    # non-CUDA (CPU, for example) float16 -> float32
-    # TODO: update this when isclose is implemented for CPU float16
-    if (compare_dtype is torch.float16 and
-        (a.device != b.device or a.device.type != 'cuda' or
-            b.device.type != 'cuda')):
-        compare_dtype = torch.float32
-
-    return compare_dtype
+
+class RelaxedBooleanPair(BooleanPair):
+    def _process_inputs(self, actual, expected, *, id):
+        # We require only one of the inputs of the inputs to be a boolean and the other can also be a boolean or
+        # a single element tensor or array, whereas in default BooleanPair both inputs have to be booleans.
+        if not (
+            (
+                isinstance(actual, self._supported_types)
+                and isinstance(expected, (*self._supported_types, torch.Tensor, np.ndarray))
+            )
+            or (
+                isinstance(expected, self._supported_types)
+                and isinstance(actual, (*self._supported_types, torch.Tensor, np.ndarray))
+            )
+        ):
+            raise UnsupportedInputs()
+
+        return [self._to_bool(input, id=id) for input in (actual, expected)]
+
+    def _to_bool(self, bool_like, *, id):
+        if isinstance(bool_like, (torch.Tensor, np.ndarray)):
+            numel = bool_like.numel() if isinstance(bool_like, torch.Tensor) else bool_like.size
+            if numel > 1:
+                raise ErrorMeta(
+                    ValueError,
+                    f"Only single element tensor-likes can be compared against a boolean. "
+                    f"Got {numel} elements instead.",
+                    id=id,
+                )
+
+            return bool(bool_like.item())
+        else:
+            return super()._to_bool(bool_like, id=id)
+
+
+class RelaxedNumberPair(NumberPair):
```

Comment:
Same question -- could we get away without supporting these relaxes variants?

Reply:
See https://github.com/pytorch/pytorch/pull/67796#discussion_r750019366.